.github/workflows/heroku_deploy.yml

```diff
@@ -0,0 +1,50 @@
+# This is a basic workflow to help you get started with Actions
+name: Dev deploy
+# Controls when the workflow will run
+on:
+  # Runs on the push to the develop branch
+  push:
+    branches:
+      - develop
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  build:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
+      - uses: actions/checkout@v2
+      # setup the node development environment
+      - uses: actions/setup-node@v2
+        with:
+          node-version: "16.x"
+      # Runs a single command using the runners shell
+      - name: Install all dependencies
+        run: npm install
+
+      # Runs a set of commands using the runners shell
+      # to added after the merge of test PR
+      - name: Run tests
+        run: npm run test
+  deploy:
+    # deploy to heroku dev pipeline
+    concurrency: 
+      group: heroku_deployment
+      cancel-in-progress: false
+    runs-on: ubuntu-latest
+    environment: development
+    needs: [build]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: akhileshns/heroku-deploy@v3.12.12
+        with:
+          heroku_api_key: ${{secrets.HEROKU_API_KEY}}
```

Comment:
how are we accessing this key? is that value a variable?

Reply:
That is a valuable stored in GitHub repository secretes. Think of is a .env file which is not accessible to the public but accessible through a runner (Workflow action)